legacy/src/main/java/org/opensearch/sql/legacy/query/DefaultQueryAction.java

```diff
@@ -100,7 +102,19 @@ public void checkAndSetScroll() {
           .getNumericalMetric(MetricName.DEFAULT_CURSOR_REQUEST_COUNT_TOTAL)
           .increment();
       Metrics.getInstance().getNumericalMetric(MetricName.DEFAULT_CURSOR_REQUEST_TOTAL).increment();
-      request.setSize(fetchSize).setScroll(timeValue);
+      request.setSize(fetchSize);
+      // Set scroll or search after for pagination
+      if (LocalClusterState.state().getSettingValue(SQL_PAGINATION_API_SEARCH_AFTER)) {
+        // search after requires results to be in specific order
+        // set sort field for search_after
+        boolean ordered = select.isOrderdSelect();
+        if (!ordered) {
+          request.addSort(FieldSortBuilder.DOC_FIELD_NAME, SortOrder.ASC);
+        }
+        // Request also requires PointInTime, but we should create pit while execution.
```

Comment:
Is this addressed?

Reply:
I think this is just a comment for the reader to call out executors handle the creation of PIT when we do this check here.